File: WingedPlate.Domain/Entities/UserEntity.cs
namespace WingedPlate.Domain.Entities;

public class UserEntity : EntityBase
{
    public string UserName { get; set; } = string.Empty;

    public string Email { get; set; } = string.Empty;

    public int RoleId { get; set; } = 0;

    public RoleEntity RoleName { get; set; } = new();   

    public byte[]? PasswordHash { get; set; }

    public byte[]? PasswordSalt { get; set; }

}

File: WingedPlate/src/entities/user.entity.ts
import { CommentEntity } from "./comment.entity";
import { RoleEntity } from "./role.entity";

export class UserEntity
{
    public userName: string = "";

    public email: string = "";

    public roleId: number = 0;

    public roleName: RoleEntity = new RoleEntity();

    public passwordHash: Uint8Array = new(Uint8Array);

    public passwordSalt: Uint8Array = new(Uint8Array);

    public comment: Array<CommentEntity> = [];
}